pallets/remote-proxy/src/lib.rs

```diff
@@ -0,0 +1,468 @@
+		DidNotFindMatchingProxyDefinition,
+		/// Proxy proof not registered.
+		ProxyProofNotRegistered,
+	}
+
+	/// The remote proxy proof to prove the existence of a proxy account.
+	#[derive(core::fmt::Debug, Clone, Decode, Encode, TypeInfo, PartialEq, Eq)]
+	pub enum RemoteProxyProof<RemoteBlockNumber> {
+		/// Assumes the default proxy storage layout.
+		RelayChain { proof: Vec<Vec<u8>>, block: RemoteBlockNumber },
+	}
+
+	/// The dispatch context to keep track of registered proofs.
+	#[derive(Default)]
+	pub(crate) struct RemoteProxyContext<RemoteBlockNumber> {
+		/// The registered proofs.
+		pub(crate) proofs: Vec<RemoteProxyProof<RemoteBlockNumber>>,
+	}
+
+	#[pallet::call]
+	impl<T: Config<I>, I: 'static> Pallet<T, I> {
+		/// Dispatch the given `call` from an account that the sender is authorised on a remote
+		/// chain.
+		///
+		/// The dispatch origin for this call must be _Signed_.
+		///
+		/// Parameters:
+		/// - `real`: The account that the proxy will make a call on behalf of.
+		/// - `force_proxy_type`: Specify the exact proxy type to be used and checked for this call.
+		/// - `call`: The call to be made by the `real` account.
+		/// - `proof`: The proof from the remote chain about the existence of the proof.
+		#[pallet::call_index(0)]
+		#[pallet::weight({
+			let di = call.get_dispatch_info();
+			(WeightInfoOf::<T, I>::remote_proxy()
+				// AccountData for inner call origin accountdata.
+				.saturating_add(T::DbWeight::get().reads_writes(1, 1))
+				.saturating_add(di.weight),
+			di.class)
+		})]
+		pub fn remote_proxy(
+			origin: OriginFor<T>,
+			real: AccountIdLookupOf<T>,
+			force_proxy_type: Option<T::ProxyType>,
+			call: Box<<T as pallet_proxy::Config>::RuntimeCall>,
+			proof: RemoteProxyProof<RemoteBlockNumberOf<T, I>>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let real = T::Lookup::lookup(real)?;
+
+			Self::do_remote_proxy(who, real, force_proxy_type, *call, proof)
+		}
+
+		/// Register a given remote proxy proof in the current [`dispatch_context`].
+		///
+		/// The registered remote proof can then be used later in the same context to execute a
+		/// remote proxy call. This is for example useful when having a multisig operation. The
+		/// multisig call can use [`Self::remote_proxy_with_registered_proof`] to get an approval by
+		/// the members of the multisig. The final execution of the multisig call should be at least
+		/// a batch of `register_remote_proxy_proof` and the multisig call that uses
+		/// `remote_proxy_with_registered_proof`. This way the final approver can use a recent proof
+		/// to prove the existence of the remote proxy. Otherwise it would require the multisig
+		/// members to approve the call in [`Config::MaxStorageRootsToKeep`] amount of time.
+		///
+		/// It is supported to register multiple proofs, but the proofs need to be consumed in the
+		/// reverse order as they were registered. Basically this means last in, last out.
```

Comment:
```suggestion
		/// reverse order as they were registered. Basically this means last in, first out.
```

Reply:
Actually the sentence before was incorrect :see_no_evil: